be/src/olap/rowset/rowset_meta.cpp

```diff
@@ -94,7 +97,29 @@ void RowsetMeta::set_fs(io::FileSystemSPtr fs) {
 void RowsetMeta::to_rowset_pb(RowsetMetaPB* rs_meta_pb) const {
     *rs_meta_pb = _rowset_meta_pb;
     if (_schema) {
-        _schema->to_schema_pb(rs_meta_pb->mutable_tablet_schema());
+        std::string schema_pb_key;
+        std::shared_ptr<TabletSchemaPB> cached_schema_pb;
+        // we cannot use schema_pb cache in variant schema or schema_version not set
+        if (_schema->num_variant_columns() == 0 && _schema->schema_version() > 0) {
```

Comment:
maybe we could capsulate the logical whether to use schema cache or not in a schema_cache function